feat: checkbox component

machikoyasuda opened the pull request:
Resolves #64
Impact: **breaking|critical|major|minor**  
Type: **feature|bugfix|performance|test|style|refactor|docs|chore**

<!-- 📦🚀 This project is deployed with [semantic-release](https://github.com/semantic-release/semantic-release) -->

<!-- Any PR with commits that start with `feat:` or `fix:` will trigger new major or minor release respectively -->

## Component
Description of the issue this component this PR adds. Also include, as necessary:
- Any other components this component requires
- Any NPM dependencies added

## Screenshots
Include mobile and desktop screenshots.

## Breaking changes
You should almost never include "BREAKING CHANGES" because we’re duplicating components to avoid that. Consult with others before doing it.

## Testing
1. List the steps needed for testing your change in this section.
2. Assume that testers already know how to start the app, and do the basic setup tasks.
3. Be detailed enough that someone can work through it without being too granular

More detail for what each of these sections should include are available in our [Contributing Docs](https://docs.reactioncommerce.com/reaction-docs/master/contributing-to-reaction)

nnnnat commented:
@machikoyasuda you've got a failing test because it can't find "lodash.uniqueid" module, looks like it needs to be added as a dependency to the `package/package.json` file. Also doesn't look like the staging link is working with this PR

machikoyasuda commented:
@nnnnat Tests are passing now thank you 💯

nnnnat commented on package.json:
```diff
@@ -94,6 +94,7 @@
     "suiteName": "jest-tests"
   },
   "dependencies": {
+    "lodash.uniqueid": "^4.0.1",
```
this dep needs to be in the `package/package.json` since it's a dependency of the components and not of styleguidest.

nnnnat commented on package/src/components/Checkbox/v1/Checkbox.js:
```diff
@@ -0,0 +1,165 @@
+`;
+
+class Checkbox extends Component {
+  static propTypes = {
```
Missing proptype comments

nnnnat commented on package/src/components/Checkbox/v1/Checkbox.js:
```diff
@@ -0,0 +1,165 @@
+  &::after {
+    content: "\f00c";
+    display: inline-block;
+    font-family: FontAwesome;
```
Could this become a theme variable?

machikoyasuda commented:
@nnnnat Ready for rereview!

machikoyasuda commented:
Blocking bug discovered by @kieckhafer:

@nnnnat, @rymorgan and I must all have FontAwesome installed locally, because, when you don't have that font locally, the checkmark renders like this:

<img width="162" alt="screen shot 2018-07-20 at 11 45 24 am" src="https://user-images.githubusercontent.com/3673236/43019599-685d661e-8c12-11e8-9349-99f4db0666c8.png">

We can do at least 2 things:
1. Use a unicode check mark instead, like http://graphemica.com/%E2%9C%94 ✔✔✔
2. Add installation instructions to make sure devs using this component install and require FontAwesome, and then, add FontAwesome to the component-library web app.

Thoughts? @aldeed @rymorgan @nnnnat

machikoyasuda commented:
After chatting with @rymorgan and checking out how @nnnnat implemented a lot of icons (complex icons!) for TextInput, @rymorgan and I decided to continue to stick with FontAwesome for the time being.

I corrected the way I was calling FontAwesome -- `Font Awesome 5 Free` instead of `FontAwesome`. I went into the Font Book app on my OS and clicked the `Disable font` option on my local copy of `FontAwesome`.

I also added ` -webkit-font-smoothing: antialiased;` because the FontAwesome docs recommend doing this when using FontAwesome in pseudoelements: https://fontawesome.com/how-to-use/on-the-web/advanced/css-pseudo-elements

Lastly, I added installing FontAwesome instructions in the docs (and the README).

cc @nnnnat

rc-publisher commented:
:tada: This PR is included in version 0.10.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@reactioncommerce/components)
- [GitHub release](https://github.com/reactioncommerce/reaction-component-library/releases/tag/v0.10.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
